mozilla/fx-private-relay-add-on
# Feature: Reuse aliases with in-page menu

## Summary

This PR refactors the entire "in-page" experience. Users can now re-use previously created, injecting the email mask address into identified forms (the same as if they generated a new mask). 

⚠️  _Note that there's an additional facet to this feature that is NOT covered in this PR, which is highlighting/promoting masks that have an association with the current site._ 

Figma design: https://www.figma.com/file/iz6X5c6IwuU6ii9128YFmP/Premium-Relay-Add-On?node-id=0%3A1 

Design caveats/revisions to note: 
- Free: There’s an inconsistency with the label for other masks: “or select from…” vs “Additional email masks”. I just use the second label the entire time.
- Premium: There’s no longer the “premium” in the logo. Eduardo and I discussed spacing constraints, so we’ve avoided it all together.
- Nebula design system: If there's any references to the design system that conflicted with this design, I deferred to Nebula where possible. 

TODO: 

- [x] BUG: Local label storage does not set/read item label correctly 
- [x] BUG: "Manage" link not accessible via keyboard tab key. (Can access it via up/down)
- [x] Keyboard a11y Check
- [x] BUG: Popup can be cut off based on where the screen alignment is. 
- [x] Add [Nebula matching Spinner animation](https://www.figma.com/file/cBaBmcPisqZ0QQX5LLzpjU/Nebula-Design-System-V1?node-id=205%3A47) 
- [x] Confirm all language uses "mask" instead of "alias"
- [x] Check functionality for users who do NOT sync with the server

## Testing (WIP) 

There are a few variant use-cases which determine the UI/functionality: 
- Amount of masks created (From zero to +5) 
- Free/premium tier 
- Label syncing with the server/local storage
- ~Are any of the masks associated with the current/active site?~

To test, start with an empty, free account. 

States to observe: 

- Not logged in
- Logged in, free account, no masks created
- Logged in, free account, one aliases created
- Logged in, free account, five masks created
- Logged in, premium account, no masks created
- Logged in, premium account, one mask created
- Logged in, premium account, six masks created

Extras: 
- Have a mixture of labels set and not set
- Try alt-clicking during this process on email input fields and observe the context menu logic still working.
